File: src/main/kotlin/com/annotation/log/logutil/util/LogKey.java
package com.annotation.log.logutil.util;

import java.util.UUID;

/**
 * @ClassName LogKey
 * @Description TODO
 * @Author pei.zhang
 * @Date 2018/10/23
 * @Version 1.0
 */
public class LogKey {
    private static final ThreadLocal<String> threadLocal = new ThreadLocal();

    public static String getThreadKey() {
        String key = threadLocal.get();
        if (null == key) {
            key = getLogKey();
            set(key);
        }
        return key;
    }

    public static void clearThreadKey() {
        threadLocal.remove();
    }

    private static void set(String key) {
        threadLocal.set(key);
    }

    private static String getLogKey() {
        return "K" + UUID.randomUUID().toString().toUpperCase().replace("-", "").substring(0, 12);
    }
}

File: src/main/kotlin/com/annotation/log/logutil/util/LogKey.kt
@file:JvmName("LogKey")
package com.annotation.log.logutil.util

import com.annotation.log.logutil.domain.BaseInfo
import com.annotation.log.logutil.domain.initBaseInfo
import java.util.*

/**
 * @ClassName   LogKey
 * @Description 日志关键字
 * @Author      pei.zhang
 * @Date        2018/10/18
 * @Version     1.0
 */
private val threadLocal = ThreadLocal<BaseInfo>()

fun GetThreadKey(): String {
    var info: BaseInfo? = threadLocal.get()
    if (null == info) {
        info = initBaseInfo(getLogKey())
        set(info)
    }
    return info.threadKey
}

fun ClearThreadKey() {
    threadLocal.remove()
}

private fun set(base: BaseInfo) {
    threadLocal.set(base)
}

private fun getLogKey(): String {
    return "K" + UUID.randomUUID().toString().toUpperCase().replace("-", "").substring(0, 12)
}